nfstream/meter.py

```diff
@@ -441,12 +466,28 @@ def meter_workflow(
             )
             return
 
+        index = 0
         remaining_packets = True
         while remaining_packets:
+            index += 1
             nf_packet = ffi.new("struct nf_packet *")
-            ret = lib.capture_next(
-                capture, nf_packet, decode_tunnels, n_roots, root_idx, int(mode)
-            )
+            if mode == NFMode.MP_QUEUE:
+                try:
+                    ts, buf = source.get(timeout=MP_QUEUE_TIMEOUT)
+                    ts_ms = int(ts * TICK_RESOLUTION)
+                    cap_length = len(buf)
+                    length = len(buf)
+                    ret = lib.consume_next(
+                        capture, nf_packet, decode_tunnels, n_roots, root_idx, int(mode),
+                        ts_ms, cap_length, length, buf
+                    )
+                except queue.Empty:
+                    ret = -2
```

Comment:
How do you plan to handle this error code at the bottom of the loop? `-1` is handled as an empty buffer and the loop continues normally, however `-2` results in an `end of file` kind of error, which terminates the loop prematurely for an empty queue.